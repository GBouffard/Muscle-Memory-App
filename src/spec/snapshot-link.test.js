import React from 'react';
import SnapShotLink from '../snapshot_link';
import renderer from 'react-test-renderer';

describe('Link ', () => {
  const component = renderer.create(
    <SnapShotLink
      page="http://www.facebook.com">
      Facebook / Doesnt have to be the same as the app
    </SnapShotLink>
  );
  let tree = component.toJSON();

  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  it('changes the class when hovered', () => {
    tree.props.onMouseEnter();
    // snaposhot need to be re-rendered when action has been performed
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('changes the class when left', () => {
    tree.props.onMouseLeave();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});