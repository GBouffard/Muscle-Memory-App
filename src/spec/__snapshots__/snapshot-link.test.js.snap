exports[`Link  changes the class when hovered 1`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Facebook / Doesnt have to be the same as the app
</a>
`;

exports[`Link  changes the class when left 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Facebook / Doesnt have to be the same as the app
</a>
`;

exports[`Link  renders correctly 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Facebook / Doesnt have to be the same as the app
</a>
`;
